module/__init__.py

```diff
@@ -18,21 +18,39 @@
 import ctypes.util
 import functools
 import io
+import os
 import platform
 import struct
 import weakref
 
-from .ffi import ffi
-
-if platform.system() == "Darwin":
-    soname = "libxcb.dylib"
-elif platform.system() == "Windows":
-    soname = "libxcb.dll"
-else:
-    soname = ctypes.util.find_library("xcb")
-    if soname is None:
-        soname = "libxcb.so"
-lib = ffi.dlopen(soname)
+
+# Attempt api mode if installed
+api_mode = True
+if ('XCFFIB_API_MODE' in os.environ and
+        int(os.environ['XCFFIB_API_MODE']) == 0):
```

Comment:
I wonder: can we do something clever where we detect the api mode vs. abi mode, so applications that use xcffib don't always have to export this environment variable in tandem with how it was installed?

Maybe something like:
```
try:
    from xcffib.ffi_build import ffi
except ImportError:
    from xcffib.ffi import ffi
```

So that downstream users can always just use our re-exported copy of ffi?

Reply:
Agreed - in fact the env vars are checked only to explicitly disable rather than needing to be set to work - I use a fallback, but could do the same for xcffib  too - let me look at tidying that up.

Reply:
I've removed the env checks at runtime and preserved them only for install - so if the lib is available in api mode it will be used.  I'll do similarly in downstream dependencies.